src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,123 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75d;
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+    }
+
+    private int capacity = DEFAULT_INITIAL_CAPACITY;
+    private int threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private int size;
+    private int index;
+
+    private Node<K, V>[] table = (Node<K, V>[]) new Node[capacity];
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> existedNode = searchHashMap(key);
+        if (existedNode == null) {
+            newHashMapNode(key, value);
+        } else {
+            existedNode.value = value;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> existedNode;
+        return (existedNode = searchHashMap(key)) == null ? null : existedNode.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(Object key) {
```

Comment:
```suggestion
    private int getIndex(Object key) {
```

названия методом должны начинаться с глаголов